### ISSUE TITLE
DistributedTests: UnExpected error occured during test execution. Try again.

### ISSUE THREAD

**Issue by Selvaraj1989** [Microsoft/vsts-tasks#2809]
Hi,

I have tried to run the Coded UI tests using "Test Plan" feature in Run Functional Test task. But I am getting the below error:

2016-10-07T00:24:16.1192690Z DistributedTests: Test run '4280' is in 'InProgress' state.
2016-10-07T00:24:16.1202668Z DistributedTests: Total Tests : 4, Passed Tests : 0
2016-10-07T00:24:26.1825503Z DistributedTests: Test run '4280' is in 'Aborted' state.
2016-10-07T00:24:26.1835512Z DistributedTests: Total Tests : 4, Passed Tests : 0
2016-10-07T00:24:36.1858375Z ##[warning]DistributedTests: Test run is aborted. Logging details of the run logs.
2016-10-07T00:24:36.3118351Z ##[warning]DistributedTests: New test run created.
2016-10-07T00:24:36.3128369Z ##[warning]Test Run queued for Project Collection Build Service (TEAM FOUNDATION).
2016-10-07T00:24:36.3138357Z ##[warning]DistributedTests: UnExpected error occured during test execution. Try again.
2016-10-07T00:24:36.3138357Z ##[warning]DistributedTests: Test run aborted. Test run id: 4280
2016-10-07T00:24:36.3208294Z ##[error]The test run was aborted, failing the task.
2016-10-07T00:24:36.3498338Z Finishing task: RunVisualStudioTestsusingTestAgent_

Could you please help me to resolve the issue.

Thanks in Advance
Selvaraj C

**Comment by allendm-msft** [comment]
@Selvaraj1989 can you share the logs from c:\dtalogs present on your target test machine

**Comment by Selvaraj1989** [comment]
@allendm-msft please find attached the dtalogs from my test agent machine

[DTAExecutionHost.txt](https://github.com/Microsoft/vsts-tasks/files/516732/DTAExecutionHost.txt)

Thanks
Selvaraj C

**Comment by allendm-msft** [comment]
This does not have any failures. Could you rerun your scenario and verify that the logs have an error and then send it across.
If these logs dont have any errors then check the ones in the _diag folder for errors.

**Comment by Selvaraj1989** [comment]
Hi,

Could you please tell me location of _diag folder.

Thanks
Selvaraj C

**Comment by allendm-msft** [comment]
This is a sibdir in your build agent install/xcopy location

**Comment by Selvaraj1989** [comment]
Thank you and now i am able to run the tests using "Test Plan" filter after clearing the "Cache" folder contents from build agent and test agent. But, I have another issue now : when I tried to run a test suite with 4 to 10 test cases, it is running without any issues and when I tried to run a test suite with 25 to 30 test cases, only few got running and others are shown as "Pending" status.

Is it something else I need to do to resolve this issue? Please suggest.

Thanks
Selvaraj C

**Comment by allendm-msft** [comment]
@Selvaraj1989 make sure that all your test cases have associated automation and none with automation status as planned/manual etc. Once you have all automated test cases you shouldnt have any other issue.

**Comment by allendm-msft** [comment]
@Selvaraj1989 are you unblocked now?

**Comment by Selvaraj1989** [comment]
@allendm-msft  Sorry for the late reply. I have checked all my test cases are associated and also the status are 'Automated" but still I am not able to get all the test cases executed via "Test Plan" filter.

Please let me know if any other configuration I need to check to resolve my issue.

Thanks
Selvaraj C

**Comment by allendm-msft** [comment]
Can you help us with a repro? All details about the type of tests/code/setup etc will help.

**Comment by SatishThareja** [comment]
@Selvaraj1989 Please share the logs at 'C:\DtaLogs' for this issue (clearing the older logs first).
Additionally please verify a few things:
- Test Case Filter input(available in Test Assembly mode of execution) is empty
- Execute the tests(pending ones) on the test agent machine manually using vstest.console.exe.

The path to vstest console will look something like: "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\CommonExtensions\Microsoft\TestWindow\vstest.console.exe".

**Comment by Selvaraj1989** [comment]
@SatishThareja While trying to execute after verifying the above things, I am getting the below error message shown in the build. Could you please help me to resolve this.

![build_error](https://cloud.githubusercontent.com/assets/22670803/19540537/dc5d0876-967d-11e6-88dd-dbf1d6c47b5b.jpg)

thanks
selvaraj c

**Comment by SatishThareja** [comment]
@Selvaraj1989 Can you let us know the following details:
- TFS version
- Build Agent Version
- Test Agent Version

Please attach build logs for this issue as well.

**Comment by Selvaraj1989** [comment]
@SatishThareja 
All TFS, Build Agent and Test Agent are 2015 Update 3 version.

Thanks
Selvaraj C

**Comment by SatishThareja** [comment]
@Selvaraj1989 Please share the version number of the build agent. Build agent version is logged at the beginning of the build logs.
If possible, please share build logs as well.

**Comment by Selvaraj1989** [comment]
@SatishThareja Attached the Build logs. Please let me know if any other info is required
[1_Build.txt](https://github.com/Microsoft/vsts-tasks/files/548150/1_Build.txt)

**Comment by SatishThareja** [comment]
@Selvaraj1989 Can you check the build agent version from agent pool and share the same with us?

Also, try using the following agent https://github.com/Microsoft/vsts-agent/releases/download/v2.108.0/vsts-agent-win7-x64-2.108.0.zip and let us know if this resolves your issue.

**Comment by Selvaraj1989** [comment]
@SatishThareja the build agent version is 1.95.3. I will give a try using the test agent you provided and will update you in a while.

Apart from the above issue, I have encountered the below issues as well:

1. DTAExecutionHost.exe has stopped working issue
![capture](https://cloud.githubusercontent.com/assets/22670803/19682438/35310dc0-9acb-11e6-972b-f7365cc3d4cd.JPG)

2. While trying to deploy the test agent in one of another VM (which has the same config of test agent which is working fine), the below error occurred:

_**The client cannot connect to the destination specified in the request. Verify that the service on the destination is running and is accepting requests. Consult the logs and documentation for the WS-Management service running on the destination, most commonly IIS or WinRM. If the destination is the WinRM service, run the following command on the destination to analyze and configure the WinRM service: "winrm quickconfig". For more information, see the about_Remote_Troubleshooting Help topic.**_

Also attached the build logs for the same.
[1_Build.txt](https://github.com/Microsoft/vsts-tasks/files/550319/1_Build.txt)

3. While deploying the test agent, the _**Configure Test Agent**_ is executing for long time as shown below and attached the logs for the same as well:

_**2016-10-24T01:50:03.2248967Z DistributedTests: Task 'ConfigureTestAgent' on machine 'EP-PLQAAUTO:5985' is taking time. Please Wait
2016-10-24T01:51:03.2253102Z DistributedTests: Task 'ConfigureTestAgent' on machine 'EP-PLQAAUTO:5985' is taking time. Please Wait
2016-10-24T01:52:03.2247297Z DistributedTests: Task 'ConfigureTestAgent' on machine 'EP-PLQAAUTO:5985' is taking time. Please Wait
2016-10-24T01:53:03.2251459Z DistributedTests: Task 'ConfigureTestAgent' on machine 'EP-PLQAAUTO:5985' is taking time. Please Wait
2016-10-24T01:54:03.2246008Z DistributedTests: Task 'ConfigureTestAgent' on machine 'EP-PLQAAUTO:5985' is taking time. Please Wait
2016-10-24T01:55:03.2240754Z DistributedTests: Task 'ConfigureTestAgent' on machine 'EP-PLQAAUTO:5985' is taking time. Please Wait
2016-10-24T01:56:03.2234973Z DistributedTests: Task 'ConfigureTestAgent' on machine 'EP-PLQAAUTO:5985' is taking time. Please Wait
2016-10-24T01:57:03.2239206Z DistributedTests: Task 'ConfigureTestAgent' on machine 'EP-PLQAAUTO:5985' is taking time. Please Wait
2016-10-24T01:58:03.2234012Z DistributedTests: Task 'ConfigureTestAgent' on machine 'EP-PLQAAUTO:5985' is taking time. Please Wait
2016-10-24T01:59:03.2238065Z DistributedTests: Task 'ConfigureTestAgent' on machine 'EP-PLQAAUTO:5985' is taking time. Please Wait
2016-10-24T02:00:03.2242591Z DistributedTests: Task 'ConfigureTestAgent' on machine 'EP-PLQAAUTO:5985' is taking time. Please Wait
2016-10-24T02:01:03.2237108Z DistributedTests: Task 'ConfigureTestAgent' on machine 'EP-PLQAAUTO:5985' is taking time. Please Wait
2016-10-24T02:02:03.2241859Z DistributedTests: Task 'ConfigureTestAgent' on machine 'EP-PLQAAUTO:5985' is taking time. Please Wait
2016-10-24T02:03:03.2245534Z DistributedTests: Task 'ConfigureTestAgent' on machine 'EP-PLQAAUTO:5985' is taking time. Please Wait
2016-10-24T02:04:03.2249774Z DistributedTests: Task 'ConfigureTestAgent' on machine 'EP-PLQAAUTO:5985' is taking time. Please Wait
2016-10-24T02:05:03.2254339Z DistributedTests: Task 'ConfigureTestAgent' on machine 'EP-PLQAAUTO:5985' is taking time. Please Wait
2016-10-24T02:06:03.2258750Z DistributedTests: Task 'ConfigureTestAgent' on machine 'EP-PLQAAUTO:5985' is taking time. Please Wait
2016-10-24T02:07:03.2253420Z DistributedTests: Task 'ConfigureTestAgent' on machine 'EP-PLQAAUTO:5985' is taking time. Please Wait
2016-10-24T02:08:03.2248539Z DistributedTests: Task 'ConfigureTestAgent' on machine 'EP-PLQAAUTO:5985' is taking time. Please Wait**_

[1_Build.txt](https://github.com/Microsoft/vsts-tasks/files/550350/1_Build.txt)

Please help me to resolve my issues

Thanks
Selvaraj C

**Comment by SatishThareja** [comment]
@Selvaraj1989 
For issue 1:
If you are observing the crash after test agent machine reboot, follow the steps below to resolve the issue -
Download the DtaExecutionHost executable shared @ https://1drv.ms/u/s!Aj_8waqf3hLypyxtiQw9DoD6PG5E .
1. Navigate to ‘C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE’
2. Take backup of DtaExecutionHost.exe
3. Replace the executable with the patched exe


Issue 2:
Check this link: https://technet.microsoft.com/en-us/library/ff700227.aspx

Issue 3:
Check this link: https://github.com/Microsoft/vsts-tasks/issues/2239

**Comment by Selvaraj1989** [comment]
Hi Sathish,

Sorry for late response as I went on sick leave.

I will update you on the latest status on the above issues in couple of days.

Thanks
Selvaraj C

**Comment by SatishThareja** [comment]
@Selvaraj1989 Any update from your side?

**Comment by SatishThareja** [comment]
Closing this issue for now. Will reopen if required.
